lisa/sut_orchestrator/azure/features.py

```diff
@@ -783,3 +787,117 @@ def get_azure_disk_type(disk_type: schema.DiskType) -> str:
+
+            avail_eligible_intersect.append(new_vm_size)
+
+        current_network_interface = current_vm_size.capability.network_interface
+        assert_that(current_network_interface).described_as(
+            "current_network_interface is not of type "
+            "NetworkInterfaceOptionSettings."
+        ).is_instance_of(schema.NetworkInterfaceOptionSettings)
+        current_data_path = current_network_interface.data_path  # type: ignore
+        current_core_count = current_vm_size.capability.core_count
+        assert_that(current_core_count).described_as(
+            "Didn't return an integer to represent the current VM size core count."
+        ).is_instance_of(int)
+        # Loop removes candidate vm sizes if they can't be resized to or if the
+        # change in cores resulting from the resize is undesired
+        for candidate_size in avail_eligible_intersect[:]:
+            candidate_network_interface = candidate_size.capability.network_interface
+            assert_that(candidate_network_interface).described_as(
+                "candidate_network_interface is not of type "
+                "NetworkInterfaceOptionSettings."
+            ).is_instance_of(schema.NetworkInterfaceOptionSettings)
+            candidate_data_path = candidate_network_interface.data_path  # type: ignore
+            # Can't resize from an accelerated networking enabled size to a size where
+            # accelerated networking isn't enabled
+            if (
+                schema.NetworkDataPath.Sriov in current_data_path  # type: ignore
+                and schema.NetworkDataPath.Sriov not in candidate_data_path  # type: ignore # noqa: E501
+            ):
+                # Removing sizes without accelerated networking capabilities
+                # if original size has it enabled
+                avail_eligible_intersect.remove(candidate_size)
+                continue
+
+            candidate_core_count = candidate_size.capability.core_count
+            assert_that(candidate_core_count).described_as(
+                "Didn't return an integer to represent the "
+                "candidate VM size core count."
+            ).is_instance_of(int)
+            # Removing vm size from candidate list if the change in core count
+            # doesn't align with the ResizeAction passed into this function
+            if (
+                resize_action == ResizeAction.IncreaseCoreCount
+                and candidate_core_count <= current_core_count  # type: ignore
+                or resize_action == ResizeAction.DecreaseCoreCount
+                and candidate_core_count >= current_core_count  # type: ignore
+            ):
+                avail_eligible_intersect.remove(candidate_size)
+
+        # Choose random size from the list to resize to
+        index = randint(0, len(avail_eligible_intersect) - 1)
+        resize_vm_size = avail_eligible_intersect[index]
+        self._node.capability = schema.load_by_type(
```

Comment:
update the capability after the resize is actual done.

Reply:
done